src/shared/condition.c

```diff
@@ -177,6 +178,31 @@ static int condition_test_user(Condition *c) {
         return id == getuid() || id == geteuid();
 }
 
+static int condition_test_control_group_controller(Condition *c) {
+        int r;
+        CGroupMask system_mask, wanted_mask = 0;
+
+        assert(c);
+        assert(c->parameter);
+        assert(c->type == CONDITION_CONTROL_GROUP_CONTROLLER);
+
+        r = cg_mask_supported(&system_mask);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to determine supported controllers: %m");
+
+        r = cg_mask_from_string(c->parameter, &wanted_mask);
+
+        if (r < 0 || wanted_mask <= 0) {
+                // This won't catch the case that we have an unknown controller
+                // mixed in with valid ones -- these are only assessed on the
+                // validity of the valid controllers found.
```

Comment:
nitpick, please use `/* */` comments, as per CODING_STYLE... we usually leave `//` comments for local, temporary, uncommitable changes, and only use `/* */` for the code comments that belong in the comitted tree...